iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/bucket/projectorg/DocumentOrganizationCombinerTest.java

```diff
@@ -0,0 +1,193 @@
+package eu.dnetlib.iis.wf.affmatching.bucket.projectorg;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.Function;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import eu.dnetlib.iis.wf.affmatching.bucket.projectorg.DocumentOrganizationCombiner;
+import eu.dnetlib.iis.wf.affmatching.bucket.projectorg.model.AffMatchDocumentOrganization;
+import eu.dnetlib.iis.wf.affmatching.bucket.projectorg.model.AffMatchDocumentProject;
+import eu.dnetlib.iis.wf.affmatching.bucket.projectorg.model.AffMatchProjectOrganization;
+import scala.Tuple2;
+
+/**
+ * @author mhorst
+ *
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class DocumentOrganizationCombinerTest {
+
+    DocumentOrganizationCombiner builder = new DocumentOrganizationCombiner();
+
+    @Mock
+    private JavaRDD<AffMatchDocumentProject> documentProjectRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<AffMatchDocumentProject, String>> keyByDocumentProjectFunction;
+
+    @Mock
+    private JavaPairRDD<String, AffMatchDocumentProject> documentProjectPairedRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<Tuple2<String, AffMatchDocumentProject>, Boolean>> filterDocumentProjectFunction;
+
+    @Mock
+    private JavaPairRDD<String, AffMatchDocumentProject> documentProjectPairedFilteredRdd;
+
+    @Mock
+    private JavaRDD<AffMatchProjectOrganization> projectOrganizationRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<AffMatchProjectOrganization, String>> keyByProjectOrganizationFunction;
+
+    @Mock
+    private JavaPairRDD<String, AffMatchProjectOrganization> projectOrganizationPairedRdd;
+
+    @Mock
+    private JavaPairRDD<String, Tuple2<AffMatchDocumentProject, AffMatchProjectOrganization>> documentOrganizationJoinedRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<Tuple2<String, Tuple2<AffMatchDocumentProject, AffMatchProjectOrganization>>, AffMatchDocumentOrganization>> mapDocProjJoinedWithProjOrgToDocOrgFunction;
+
```

Comment:
Well, yes now it is a sort of a monster name :-)) Would you mind changing the name to smth simpler like createDocOrgMapFunction or something like this... ;-)